WooCommerce/Classes/ViewRelated/Search/SearchCommand.swift

```diff
@@ -0,0 +1,33 @@
+import UIKit
+
+/// An interface for search UI associated with a generic model and cell view model.
+protocol SearchCommand {
+    associatedtype Model
+    associatedtype CellViewModel
+
+    /// The placeholder of the search bar.
+    var searchBarPlaceholder: String { get }
+
+    /// Displayed when there are no search results.
+    var emptyStateText: String { get }
+
+    /// Creates a view model for the search result cell.
+    ///
+    /// - Parameter model: search result model.
+    /// - Returns: a view model based on the search result model.
+    func createCellViewModel(model: Model) -> CellViewModel
```

Comment:
Do you think we really need this? I know we have been passing view models to cells in the past, but if you got rid of this cell view models and let the cell manipulate the model object directly, you could get rid as well of the associated type, which would also allow you to make SearchViewController generic only on model object and cell, requiring one less generic type. What do you think?

Reply:
the reason for transforming `Model` (`Order`) to a view model for the search result cell is because for Order cell, it needs not only `Order` but `OrderStatus` from another status results controller (owned by the Order search command). do you have other ideas to achieve this, passing data other than `Model` to configure a search result cell?

Reply:
argh. There isn't much we can do then. Shall we convince ourselves, then, that it's a good thing to have an extra layer of indirection here, in the form of a cell view model? 😛

Reply:
yea, ideally the search result cell can be rendered by just the model but not in Order cell's case 😐